File: core/src/com/spacefist/managers/EnemyMineManager.java
package com.spacefist.managers;

import com.badlogic.gdx.math.Vector2;
import com.spacefist.GameData;
import com.spacefist.entities.EnemyMine;

public class EnemyMineManager extends Manager<EnemyMine> {

    public EnemyMineManager(GameData gameData) {
        super(gameData);
        this.gameData = gameData;
    }

    public void SpawnEnemyMine(int x, int y) {
        Add(new EnemyMine(gameData, new Vector2(x, y)));
    }
}

File: core/src/com/spacefist/managers/EnemyMineManager.kt
package com.spacefist.managers

import com.badlogic.gdx.math.Vector2
import com.spacefist.GameData
import com.spacefist.entities.EnemyMine

class EnemyMineManager(gameData: GameData) : Manager<EnemyMine>(gameData) {

    init {
        this.gameData = gameData
    }

    fun spawnEnemyMine(x: Int, y: Int) {
        add(EnemyMine(gameData, Vector2(x.toFloat(), y.toFloat())))
    }
}
